streamalert/shared/helpers/aws_api_client.py

```diff
@@ -182,3 +183,71 @@ def download_fileobj(file_handle, bucket, key, region):
         except ClientError:
             LOGGER.error('An error occurred during S3 DownloadFileobj')
             raise
+
+
+class AwsSsm:
+    """Class containing helper functions for aws ssm"""
+
+    @staticmethod
+    def get_parameter(parameter_name, region, with_decryption=True):
+        """gets the parameter_name from SSM Parameter store
+
+        Args:
+            parameter_name (str): The name of the parameter to fetch
+            region (str): AWS region
+            with_decryption (bool): Should decryption be attempted via this call
+
+        Returns:
+            str: The parameter either encrypted or not
+
+        Raises:
+            ClientError
+        """
+        client = boto3.client('ssm', config=default_config(region=region))
```

Comment:
Some boto3 clients maintain some sort of "internal state". For example, the `athena` one (I think?) performs API calls when you first instantiate it. Does SSM use this?

Alternative pattern with this class could be something like:

```
class AwsSsmClient:

  def __init__(config):
    self._client = boto3.client('ssm', config)

  def get_parameter(parameter_name, with_decryption=True):
    # ...
```

Reply:
Since this is a one time throw away script I'd argue this is good enough as is. Since after the migration is complete this can be thrown away

Reply:
The previous one had this pattern since everywhere it was called was via the `staticmethod`. i could instantiate this Class within the callers init so it is cached etc between invocations but the parameter/secret that is pulled back is already cached so this seems like complexity for complexities sake